### ISSUE TITLE
[BUG] Duplicate Disks Displayed

### ISSUE THREAD

**Issue by ibrokemypie** [aristocratos/btop#255]
**Describe the bug**

Many of my disks in the disks window are displayed multiple times.

**To Reproduce**

Open btop and add wanted disk mountpoints to the disks filter field.

**Expected behavior**

Each mountpoint is displayed once

**Screenshots**

![image](https://user-images.githubusercontent.com/7827846/152092567-7a80cd37-b8aa-47c7-b66e-7247d79eef1d.png)

![image](https://user-images.githubusercontent.com/7827846/152092616-46dbe0d6-b03c-493f-ae7e-dcb24e098e7e.png)

[If applicable, add screenshots to help explain your problem.]

**Info (please complete the following information):**
 - btop++ version: `btop -v` 1.2.0
 - Binary: [self compiled or static binary from release] Binary from arch community repo
 - (If compiled) Compiler and version:
 - Architecture: [x86_64, aarch64, etc.] `uname -m` x86_64
 - Platform: [Linux, FreeBSD, OsX] Linux
 - (Linux) Kernel: `uname -r` 5.16.4-242-tkg-bmq-llvm
 - (OSX/FreeBSD) Os release version:
 - Terminal used: foot
 - Font used: CodingFontTobi1

**Additional context**

contents of `~/.config/btop/btop.log`

```
2022/02/02 (15:24:29) | ===> btop++ v.1.2.0
2022/02/02 (15:24:29) | DEBUG: Starting in DEBUG mode!
2022/02/02 (15:24:29) | INFO: Logger set to DEBUG
2022/02/02 (15:24:29) | DEBUG: Using locale en_AU.UTF-8
2022/02/02 (15:24:29) | INFO: Running on /dev/pts/2
2022/02/02 (15:24:35) | INFO: Quitting! Runtime: 00:00:06
```

output of `mount` (my mounts are a bit unusual so I think that is where btop is getting confused)
```
dev on /dev type devtmpfs (rw,nosuid,relatime,size=8145556k,nr_inodes=2036389,mode=755,inode64)
sys on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
run on /run type tmpfs (rw,nosuid,nodev,mode=755,inode64)
efivarfs on /sys/firmware/efi/efivars type efivarfs (rw,nosuid,nodev,noexec,relatime)
/dev/sda2 on / type btrfs (rw,noatime,compress-force=zstd:3,ssd,space_cache=v2,subvolid=5,subvol=/)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev,inode64)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
bpf on /sys/fs/bpf type bpf (rw,nosuid,nodev,noexec,relatime,mode=700)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=30,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=12754)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,nosuid,nodev,noexec,relatime)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime,pagesize=2M)
tracefs on /sys/kernel/tracing type tracefs (rw,nosuid,nodev,noexec,relatime)
configfs on /sys/kernel/config type configfs (rw,nosuid,nodev,noexec,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,nosuid,nodev,noexec,relatime)
ramfs on /run/credentials/systemd-sysusers.service type ramfs (ro,nosuid,nodev,noexec,relatime,mode=700)
systemd-1 on /boot type autofs (rw,relatime,fd=46,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=14180)
systemd-1 on /data type autofs (rw,relatime,fd=47,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=14183)
systemd-1 on /games type autofs (rw,relatime,fd=48,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=14186)
systemd-1 on /home/pie/.cache/paru type autofs (rw,relatime,fd=49,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=14189)
systemd-1 on /home/pie/.cache/yay type autofs (rw,relatime,fd=50,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=14192)
systemd-1 on /home/pie/Downloads type autofs (rw,relatime,fd=51,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=14195)
systemd-1 on /vm type autofs (rw,relatime,fd=52,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=14198)
tmpfs on /tmp type tmpfs (rw,nosuid,nodev,size=8154204k,nr_inodes=1048576,inode64)
binfmt_misc on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,nosuid,nodev,noexec,relatime)
net_cls on /sys/fs/cgroup/net_cls type cgroup (rw,relatime,net_cls)
/dev/sda1 on /boot type vfat (rw,noatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro,x-systemd.automount)
/dev/sdb on /data type btrfs (rw,noatime,compress-force=zstd:3,space_cache=v2,subvolid=258,subvol=/data)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=1630840k,nr_inodes=407710,mode=700,uid=1000,gid=1000,inode64)
gvfsd-fuse on /run/user/1000/gvfs type fuse.gvfsd-fuse (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
portal on /run/user/1000/doc type fuse.portal (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
/dev/sdb on /home/pie/Downloads type btrfs (rw,noatime,compress-force=zstd:3,space_cache=v2,subvolid=258,subvol=/data,x-systemd.automount)
/dev/sdb on /games type btrfs (rw,noatime,compress-force=zstd:3,space_cache=v2,subvolid=256,subvol=/games)
/dev/sdb on /vm type btrfs (rw,noatime,compress-force=zstd:3,space_cache=v2,subvolid=257,subvol=/vm)
/dev/sdb on /home/pie/.cache/paru type btrfs (rw,noatime,compress-force=zstd:3,space_cache=v2,subvolid=258,subvol=/data,x-systemd.automount)
/dev/sdb on /home/pie/.cache/yay type btrfs (rw,noatime,compress-force=zstd:3,space_cache=v2,subvolid=258,subvol=/data,x-systemd.automount)
```

**Comment by foundObjects** [comment]
I just came to report this too, it's been an issue for btop and bpytop for a while. It looks like separate BTRFS subvolume mountpoints are detected as separate filesystems.

![Screenshot from 2022-02-06 00-23-33](https://user-images.githubusercontent.com/28817345/152673090-fa60cefe-3738-4644-b7c6-dbcc05ed8b14.png)

Personally I'd like to see only the logically highest mount point in the VFS tree reported in the disks list eg:

```
[ arglebargle@arch-zephyrus]$ findmnt -t btrfs
TARGET                                  SOURCE                               FSTYPE OPTIONS
/                                       /dev/nvme0n1p1[/Arch/@root]          btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=1039,subvol=/Arch/@root
├─/var/log                              /dev/nvme0n1p1[/Arch/var/@log]       btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=262,subvol=/Arch/var/@log
├─/home                                 /dev/nvme0n1p1[/@home]               btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=1040,subvol=/@home
│ ├─/home/arglebargle/Downloads         /dev/nvme0n1p1[/userdata/@downloads] btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=268,subvol=/userdata/@downloads
│ ├─/home/arglebargle/.cache            /dev/nvme0n1p1[/userdata/@cache]     btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=267,subvol=/userdata/@cache
│ ├─/home/arglebargle/.local/share/Steam
│ │                                     /dev/nvme0n1p1[/userdata/@steam]     btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=265,subvol=/userdata/@steam
│ └─/home/arglebargle/work              /dev/nvme0n1p1[/userdata/@work]      btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=266,subvol=/userdata/@work
├─/mnt/btrfs                            /dev/nvme0n1p1                       btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=5,subvol=/
├─/var/cache                            /dev/nvme0n1p1[/Arch/var/@cache]     btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=261,subvol=/Arch/var/@cache
├─/var/lib/flatpak                      /dev/nvme0n1p1[/var/lib/@flatpak]    btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=263,subvol=/var/lib/@flatpak
└─/var/tmp                              /dev/nvme0n1p1[/var/@tmp]            btrfs  rw,noatime,compress-force=zstd:1,ssd,space_cache=v2,autodefrag,subvolid=264,subvol=/var/@tmp
```
I'd like to see `/` listed, but not the others on the same filesystem that are mounted lower in the VFS tree.

**Comment by tonyrh** [comment]
I have this issue too, but I don't have any BTRFS volumes. The duplicate disks shown are ext4 and NTFS mounted via CIFS.

**Comment by aristocratos** [comment]
@ibrokemypie @foundObjects 
Should be fixed in latest commit. See the [continuous build](https://github.com/aristocratos/btop/actions/runs/1861155945) if you need binaries.

**Comment by tonyrh** [comment]
> @ibrokemypie @foundObjects @tonyrh Should be fixed in latest commit. See the [continuous build](https://github.com/aristocratos/btop/actions/runs/1861155945) if you need binaries.

Yes, it's fixed. Thanks!

**Comment by aristocratos** [comment]
Fixed in v1.2.4
